### PR TITLE
Updated CONTRIBUTING.md for build process

### PR DESCRIPTION
**Pull request by dmueller39**
This was what was missing based on my experiences with https://github.com/AtomLinter/linter-eslint/pull/694. Anything else I should add?

### CODE REVIEW

**Comment by dmueller39** [comment]
Actually should I be linking to this instead? https://github.com/AtomLinter/linter-eslint/wiki/Linking-to-Rule-Documentation#how-you-can-help

**Comment by IanVS** [comment]
https://github.com/AtomLinter/linter-eslint/wiki/Linking-to-Rule-Documentation#how-you-can-help is out-of-date, and I need to remove it.

It would probably be best to clarify/expand on https://github.com/AtomLinter/linter-eslint/blob/master/CONTRIBUTING.md, and link to that file from the README.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,8 +6,10 @@ If you would like to contribute enhancements or fixes, please do the following:
 
 1.  Fork the plugin repository
 2.  Run `npm install` to setup all dependencies
-3.  Hack on a separate topic branch created from the latest `master`
-4.  Commit and push the topic branch
+3.  Hack on a separate topic branch created from the latest `master`. The src files are under `/src`
+4.  Check for lint errors with `npm run lint` or Atom and `linter-eslint`.
+5.  Run `npm run compile` to generate compiled files under `lib/`.
+4.  Commit the changes under `src/` and `lib/` and push the topic branch
```
Comment (Arcanemagus) [inline CONTRIBUTING.md]: `4.` -> `6.`, also the next two lines should have their number updated.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,8 +6,10 @@ If you would like to contribute enhancements or fixes, please do the following:
 
 1.  Fork the plugin repository
 2.  Run `npm install` to setup all dependencies
-3.  Hack on a separate topic branch created from the latest `master`
-4.  Commit and push the topic branch
+3.  Hack on a separate topic branch created from the latest `master`. The src files are under `/src`
```
Comment (Arcanemagus) [inline CONTRIBUTING.md]: This second sentence is repeating itself, maybe change it to something along the lines of:

```markdown
Changes to the package code should be made to the files in the `src` directory.
```

Comment (Arcanemagus) [inline CONTRIBUTING.md]: I would remove all of the directory separators from the folder names, they aren't necessary, are inconsistently used (before v.s after), and are OS specific.

**Comment by dmueller39** [comment]
LGTM

**Pull request merged**